lib/dal/src/attribute/prototype.rs

```diff
@@ -98,7 +98,9 @@ pub enum AttributePrototypeEventualParent {
     SchemaVariantFromProp(SchemaVariantId, PropId),
 }
 
-id!(AttributePrototypeId);
+// TODO(nick): switch to the "id!" macro once the frontend doesn't use the old nil id to indicate
+// that the argument is a new one.
+pk!(AttributePrototypeId);
```

Comment:
You'll see this a few times in the diff. Essentially, the frontend indicates that the func is new by using the `nilId`. For nodes on the graph, we've scrubbed the usage of `nilId` where possible, but we need we will be friends with it for now since we are trying to touch func authoring flow as little as possible.